### PR TITLE
zh-TW: translate std

### PR DESCRIPTION
**Pull request by mgeisler**
Part of #684.

### CODE REVIEW

**Comment by mgeisler** [comment]
Hey all, please take a look and leave suggestions if there is something here I should fix.

File: po/zh-TW.po [hunk 1]
```diff
@@ -7369,15 +7391,20 @@ msgid ""
 "large amounts of data on the stack, instead store the data on the heap in a "
 "`Box` so only the pointer is moved."
 msgstr ""
+"* `Box` 就像是 C++ 中的 `std::unique_ptr`，兩者的差別在於 `Box` 不會是空值。"
+"* 上面的範例使用 `Deref`，因此 `println!` 陳述式甚至可以省略 `*`。* 在以下情"
+"況下，您可以使用 `Box`：* 編譯時遇到不知道大小為何的類型，但 Rust 編譯器需要"
+"知道確切大小。* 想要轉移大量資料的擁有權。為避免在堆疊上複製大量資料，請改將"
+"資料儲存在 `Box` 的堆積上，這樣系統就只會移動指標。"
```
Comment (johnathan79717) [inline po/zh-TW.po:7398]:
```suggestion
"* `Box` 就像是 C++ 中的 `std::unique_ptr`，兩者的差別在於 `Box` 不會是空值。\n"
"* 上面的範例使用 `Deref`，因此 `println!` 陳述式甚至可以省略 `*`。\n"
"* 在以下情況下，您可以使用 `Box`：\n"
"   * 編譯時遇到不知道大小為何的類型，但 Rust 編譯器需要知道確切大小。\n"
"   * 想要轉移大量資料的所有權。為避免在堆疊上複製大量資料，請改將"
"資料儲存在 `Box` 的堆積上，這樣系統就只會移動指標。"
```

**Review by johnathan79717 — CHANGES_REQUESTED** [review]

File: po/zh-TW.po [hunk 2]
```diff
@@ -7058,6 +7060,11 @@ msgid ""
 "* [`Rc`](std/rc.md): a shared reference-counted pointer for heap-allocated "
 "data."
 msgstr ""
+"* [`Option` 和 `Result`](std/option-result.md) 類型：用於選擇性的值和[錯誤處"
+"理](error-handling.md)。* [`String`](std/string.md)：用於自有資料的預設字串類"
+"型。* [`Vec`](std/vec.md)：標準的可延伸向量。* [`HashMap`](std/hashmap.md)："
+"採用可設定雜湊演算法的雜湊映射類型。* [`Box`](std/box.md)：堆積配置資料的自有"
+"指標。* [`Rc`](std/rc.md)：堆積配置資料的共用參考指標。"
```
[the remark this answers is not in the document]
Comment (mgeisler) [reply po/zh-TW.po:7067]: > I notice that many newlines between list items have been removed in this PR.

Yeah, I think our tooling displays the text in a way that isn't useful to the translators. So they struggle to recreate the Markdown in the translation and result is what you see. I'll go over the other PRs and fix them.

[the remark this answers is not in the document]
Comment (mgeisler) [reply po/zh-TW.po]: #1036 will fix some more instances of this (hopefully all).

File: po/zh-TW.po [hunk 3]
```diff
@@ -7317,11 +7345,11 @@ msgstr ""
 
 #: src/std/box.md:1
 msgid "# `Box`"
-msgstr ""
+msgstr "# `Box`"
 
 #: src/std/box.md:3
 msgid "[`Box`][1] is an owned pointer to data on the heap:"
-msgstr ""
+msgstr "[`Box`][1] 是指向堆積上的資料的具有所有權的指標："
```
Comment (johnathan79717) [inline po/zh-TW.po:7352]:
```suggestion
msgstr "[`Box`][1] 是具有所有權的指向堆積上的資料的指標："
```

File: po/zh-TW.po [hunk 4]
```diff
@@ -7434,17 +7470,25 @@ msgid ""
 "reference of some kind, instead of storing a value directly.   \n"
 "    "
 msgstr ""
+"* 如果這裡未使用 `Box`，且我們要將 `List` 直接嵌入 `List`，編譯器就不會計算記"
+"憶體中的結構體固定大小，結構體看起來會像是無限大。\n"
```
Comment (johnathan79717) [inline po/zh-TW.po:7474]:
```suggestion
"* 如果這裡未使用 `Box`，且我們要將 `List` 直接嵌入 `List`，編譯器就不會計算出固定的"
"結構體記憶體大小，結構體看起來會像是無限大。\n"
```

**Review by johnathan79717 — APPROVED** [review]

**Pull request merged**